code/modules/unit_tests/embedding.dm

```diff
@@ -4,11 +4,10 @@
 	var/mob/living/carbon/human/victim = allocate(/mob/living/carbon/human/consistent)
 	var/mob/living/carbon/human/firer = allocate(/mob/living/carbon/human/consistent)
 	var/obj/projectile/bullet/c38/bullet = new(get_turf(firer))
-	bullet.set_embed(bullet.get_embed().generate_with_values(embed_chance = 100))
-	TEST_ASSERT_EQUAL(bullet.get_embed().embed_chance, 100, "embed_chance failed to modify")
+	bullet.get_embed().embed_chance = 100
+	TEST_ASSERT_EQUAL(bullet.embed_data.embed_chance, 100, "embed_chance failed to modify")
```

Comment:
Ok I assume this goes from `get_embed()` to `.embed_data` because we assert that it has data now that the above proc has lazyloaded this

But I think this is messy? Like if someone comes along and changes the order of the code, it breaks

I feel like all accesses of `.embed_data` should be `.get_embed()` (and it should be private/protected)